api/v1beta1/azurecluster_default.go

```diff
@@ -146,6 +154,11 @@ func (c *AzureCluster) setSubnetDefaults() {
 			RouteTable: RouteTable{
 				Name: generateNodeRouteTableName(c.ObjectMeta.Name),
 			},
+			NatGateway: NatGateway{
+				NatGatewayClassSpec: NatGatewayClassSpec{
+					Name: fmt.Sprintf("%s-%s", c.ObjectMeta.Name, "node-natgw"),
```

Comment:
Can you move the `Sprintf` to its own helper function `generateNatGatewayName` to match the pattern above?

Reply:
Sure, this is a good idea!